gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Table.java

```diff
@@ -322,14 +322,15 @@ public Builder toBuilder() {
   }
 
   @Override
-  public boolean equals(Object obj) {
-    return obj instanceof Table
+  public final boolean equals(Object obj) {
+    return this == obj
+        || obj instanceof Table
```

Comment:
Yes, you are right.

The difference is what would you *prefer* to happen when I create a subclass (not a mock, so with real values) and call `tableInstance.equals(subclassInstance)`.
If you are fine with True in this case than keep as is, otherwise I think you should change to getClass() equality. 

To me, the fact that the class is not final is a compromise that is done (and should be done) for testing (and only for testing) and therefore would do whatever is possible to minimize errors.
However, this is not a big deal and I am fine either way.